test/16_recode.jl

```diff
@@ -534,4 +538,70 @@ end
     @test !isordered(y)
 end
 
+@testset "replace with CategoricalArray" begin
+    function testf(f, T, x::CategoricalArray, y, pairs::Pair...)
+        ca = f(x, pairs...)
+        @test ca ≅ y
+        @test ca isa CategoricalArray{T}
+
+        if VERSION >= v"0.7.0-"
+            a = f(Array(x), pairs...)
```

Comment:
Since you do this test, it would make sense to drop the `y` argument altogether. We don't really care if it's not tested on 0.6 at this point.

Also it's not very useful to test `a isa Array{T}`.

Reply:
Alright, I'll test `ca` against `a` instead on 0.7 only, and remove the `a isa Array{T}`